lib/Config/Shortcodes.php

```diff
@@ -39,18 +45,18 @@ function init() {
     $this->wp->addFilter('mailpoet_archive_subject', [
       $this, 'renderArchiveSubject',
     ], 2, 3);
-
+    // initialize subscription pages data
+    $this->subscription_pages->init();
     // initialize subscription management shortcodes
-    $subscription_page = new Pages();
-    $subscription_page->initShortcodes();
+    $this->subscription_pages->initShortcodes();
   }
 
   function formWidget($params = []) {
     // IMPORTANT: fixes conflict with MagicMember
-    WPFunctions::get()->removeShortcode('user_list');
+    $this->wp->removeShortcode('user_list');
 
     if (isset($params['id']) && (int)$params['id'] > 0) {
-      $form_widget = new \MailPoet\Form\Widget();
+      $form_widget = new Widget();
```

Comment:
Why didn't you pass Widget through DI as well?

Reply:
This class extends `\WP_Widget` and is consumed by WP via [register_widget](https://codex.wordpress.org/Function_Reference/register_widget) function. WP does not pass any parameters to a widget constructor and, moreso, our DI container throws a non-existent `WP_Widget` class error when dumping container cache during build with this class added. Therefore, even if refactoring it to DI is still possible, it would be a longer and more quirky refactoring session.